lib/os/fdtable.c

```diff
@@ -65,6 +65,13 @@ static int z_fd_unref(int fd)
 {
 	int old_rc = atomic_dec(&fdtable[fd].refcount);
 
+	/* Reference counter must be checked to avoid decrement refcount below
+	 * zero causing file descriptor leak.
+	 */
+	if (atomic_get(&fdtable[fd].refcount) < 0) {
+		atomic_set(&fdtable[fd].refcount, 0);
```

Comment:
This is no longer atomic as we have now two separate calls. We need to use `atomic_cas()` to make the check atomic too.

Reply:
Changed to `atomic_cas`. Please take a look now.